syft/generic/frameworks/hook/hook.py

```diff
@@ -407,6 +419,73 @@ def overloaded_native_method(self, *args, **kwargs):
 
         return overloaded_native_method
 
+    @classmethod
+    def _get_hooked_private_method(cls, method_name):
```

Comment:
Why was this necessary to put here? Is this not the same as https://github.com/OpenMined/PySyft/blob/f39a93d085b74f16724633fd12a28c893a724c59/syft/generic/frameworks/hook/hook.py#L365

Reply:
In the final of `_get_hooked_private_method`, we're setting pointers to the result tensor parent(s) and saving the command/operation used to generate it.

Reply:
But thinking about this, It doesn't seem the best way to implement it.

Reply:
Do you have any suggestion?

Reply:
ah nvm - yours looks good. Yes we could probably compress the code some more but this is readable so I'm happy